File: JeansStyle.DAL/Domain/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace JeansStyle.DAL.Domain.Models
{
    public class Product
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public List<ProductSize> ProductSizes { get; set; } //TODO
        public Guid CategoryId { get; set; }
        public virtual Category Categories { get; set; }
        public List<Season> Seasons { get; set; }
        public Guid GenderId { get; set; }
        public virtual Gender Gender { get; set; }
        public string Image { get; set; }
        
        [Column(TypeName = "decimal(18,4)")]
        public decimal Price { get; set; }
    }
}

File: JeansStyle.WEB/src/Classes/Product.js
import {Gender} from "./Gender";
import {Category} from "./Category";
import {ProductSize} from "@/Classes/ProductSize";

export class Product {
    constructor(
        id,
        title,
        image,
        description,
        price,
        productSizes,
        gender,
        category
    ) {
        this.id = id
        this.title = title
        this.image = image
        this.description = description
        this.price = parseFloat(price).toFixed(2)

        this.productSizes = productSizes.forEach(productSize => {
            this.productSizes.push(new ProductSize(...productSize))
        })
        
        this.gender = new Gender(...gender)
        this.category = new Category(...category)
    }
}